docs/Codebase.md

```diff
@@ -0,0 +1,96 @@
+```csharp
+UpdateResults UpdateVersions(
+            IEnumerable<string> manifestFiles,
+            string defaultConfigFile,
+            IEnumerable<Regex> whitelistedPackages,
+            ImmutableHashSet<string>? packagesToIgnore,
+            string? accessToken,
+            string? propsFilesRootDirectory,
+            Predicate<Build>? buildFilter);
+```
+- __manifestFiles:__ The paths to all the manifest.json files to be inserted.
+- __defaultConfigFile:__ Full path to _default.config_ file.
+- __whitelistedPackages:__ Regex patterns matching with IDs of the packages that are allowed to be updated. If the set is empty,
+ any package is allowed to be updated unless specified in packagesToIgnore argument.
+- __packagesToIgnore:__ IDs of the packages to be ignored.
+- __accessToken:__ Access token to be used when connecting to VS feed.
+- __propsFilesRootDirectory:__ Directory that will be searched for `.props` files.
+- __buildFilter:__ Predicate to determine if a build should be included in the insertion process.
+- Returns: an `UpdateResults` instance, detailing the outcome of the operation.
+
+In a simple case, this method can be invoked in the following way:
+```csharp
+ IInsertionApiFactory apiFactory = new InsertionApiFactory();
+ IInsertionApi api = apiFactory.Create(MaxWaitDuration, MaxDownloadDuration, MaxConcurrency);
+
+ api.UpdateVersions(new []{"\\Assets\\manifest.json"},
+    "\\\\VS\\.corext\\default.config",
+    Enumerable.Empty<Regex>(),
+    ImmutableHashSet.Create("VS.ExternalAPIs.MSBuild"),
+    null,
+    null,
+    null);
+```
+
+#### [InsertionApi Class](#insertionapi-class)
+`InsertionApi` is the only implementation of `IInsertionApi` in the solution. The main logic executes the following steps in order:
+1. Load _default.config_ file into memory. All the operations on _default.config_ are handled in `DefaultConfigUpdater` class.
+1. Load manifest.json files.
+1. Determine the updates to the _default.config_ file. This operation is executed in parallel for each asset defined in manifest files.
+1. Determine the updates to `.props` files.
+    1. Determine all the `.swr` files.
+    2. Search the contents of `.swr` files for variables of the form `$(variable_name)`.
+    3. Download the packages updated in step 3 into memory.
+    4. Determine the value of the variables by comparing the _file names in the nuget packages_ with _the file names listed in swr payloads_.
+1. In the declaring `.props` files, replace the value of the variables with the values found in the previous step.
+1. Flush changes to `.props` files.
+1. Flush changes to `default.config` and `.packageconfig` files.
+
+### [Console Project](#console-project)
+The console project contains the command line user interface codebase and logging.
+
+#### [Program Class](#program-class)
+- The Program class contains the assembly entry point
+and is the most interesting class in the assembly
```

Comment:
is line break needed here?

Reply:
Same as above, it is only displayed in IDE.